src/main/java/net/atos/entng/calendar/ical/ICalHandler.java

```diff
@@ -251,15 +252,32 @@ private void setEventDates(VEvent event, JsonObject jsonEvent) throws UnhandledE
             // End Date
             calendar.setTime(endDate);
             calendar.set(java.util.Calendar.HOUR_OF_DAY, 18);
-            calendar.add(java.util.Calendar.DATE, -1);
             endMoment = MOMENT_FORMAT.format(calendar.getTime());
             jsonEvent.put("allday", allDay);
         }
+
+        checkStartAndEndDatesOrder(startMoment, endMoment);
+
         // Put dates to jsonEvent
         jsonEvent.put("startMoment", startMoment);
         jsonEvent.put("endMoment", endMoment);
     }
 
+    private void checkStartAndEndDatesOrder(String startMoment, String endMoment) throws UnhandledEventException {
+        try {
+            java.util.Date finalStartDate = new SimpleDateFormat(DateUtils.DATE_FORMAT_UTC).parse(startMoment);
+            java.util.Date finalEndDate = new SimpleDateFormat(DateUtils.DATE_FORMAT_UTC).parse(endMoment);
+            if (Boolean.FALSE.equals(DateUtils.isStrictlyBefore(finalStartDate, finalEndDate))) {
+                String message = String.format("[Calendar@%s::setEventDates] Start date cannot be before end date",
```

Comment:
```suggestion
                String message = String.format("[Calendar@%s::setEventDates] End date cannot be before start date",
```
Ou "before" devient "after", au choix